app/routes/project/donate.js

```diff
@@ -1,12 +1,42 @@
 import Ember from 'ember';
 
 const {
-  Route
+  get,
+  inject: { service },
+  Route,
+  RSVP,
+  set
 } = Ember;
 
+const ALREADY_A_SUBSCRIBER = "You're already supporting this project.";
+
 export default Route.extend({
+  flashMessages: service(),
+  session: service(),
+  userSubscriptions: service(),
+
   model() {
-    return this.modelFor('project').reload();
+    return this.modelFor('project').reload().then((project) => {
+      if (get(this, 'session.isAuthenticated')) {
+        let subscription = get(this, 'userSubscriptions').fetchForProject(project);
+        return RSVP.hash({ project, subscription });
+      } else {
+        return RSVP.hash({ project, subscription: null });
```

Comment:
Do we use this behavior anywhere else? If so, maybe fetchForProject should just  return null if no current user.